### DISCUSSION TITLE
Is there an efficient way to know all node/edge labels?

### DISCUSSION THREAD

**Discussion by marco-brandizi** [ArcadeData/arcadedb#1109] [Q&A]
Hi all, 

I'd like to fetch all the unique node labels that are used in a DB. Similarly, I'd like to know all the edge labels (long story, I want to write a script to index a given property for all the types, eg,  `CREATE INDEX ON $type ( uri )` for each type; obviously, I'm assuming all the type instances have such a property).

I know how to do that in gremlin:

```groovy
g.V().label().dedup ()
g.E().label().dedup ()
```

But likely, this will scan all the nodes/labels and will be very slow in large databases. Or not? Is there a way that queries schema information? Similarly, are there SQL commands to query the schema, eg, SHOW ALL TABLES, DESCRIBE TABLE, etc?

**Comment by lvca** [comment]
You can execute:

```sql
select from schema:types
```

To have all the defined types.

**Reply by marco-brandizi** [reply]
Ah, now I see `schema:*` in the manual, thank you very much!
